### PR TITLE
Feature/reviews merged develop

### PR DESCRIPTION
**Pull request by mahmed0715**
------Wed Nov 20--------

- [x] In Project Single Page ( Add Write a Review Button) (if the logged in user can write a review.)
Screen Shot 2019-11-17 at 2 50 29 PM
Screen Shot 2019-11-20 at 4 14 25 AM

text `Write a Review`

_sorry, my note last time on this was confusing_

**Few UI/UX changes below:**

- [x] Swap, Heading and Sub Heading.
Screen Shot 2019-11-20 at 4 16 30 AM

- [x] Implement full stars instead of single star review. as in image above
height 19, text font size 14, semibold

- [x] Add comma after Name and before Org name
Screen Shot 2019-11-20 at 4 17 14 AM

- [x] Hide Reviews with 0 value
Screen Shot 2019-11-20 at 4 19 33 AM

- [x] Add new review to the top, old reviews at the bottom.

It doesn't seem to be fixed. as you can see date on the image above.

- [x] Implement white background like in image below.

- [ ] Implement Image display (can be done later, migrate to an issue after merging this)
Screen Shot 2019-11-20 at 4 24 39 AM

-------Update Nov 17---------

Everything looks good.
Please finish the following fixes.

- [x] Remove the current button, as the entry for reviews should be from the ratings stars.

Screen Shot 2019-11-17 at 2 51 39 PM

- [x] Currently It is not possible to remove the rating once you click the stars. On clicking the same stars again, allow to reset the value to 0, or clear the selection for review score.

If you click 5th star, = 5 stars, then clicking 4 stars = 4 stars, if click 4 stars again value =0

tldr; second click on same star resets the value. 

- [x] Rename `Community Reviews` to `Expert Reviews`

- [x] if project does not have a certified flag, make the width smaller 
Screen Shot 2019-11-17 at 2 35 12 PM

- [x] Clicking on above box, with 4.16 stars should take to list of reviews. Make at least 48px height clickable area.

- [x] Update the global review count once a review has been added, saved or updated.
Screen Shot 2019-11-17 at 2 37 37 PM

- [x] Make name of reviewer Cickable, – take the user to their tree counter.

- [x] Add new review to the top, old reviews at the bottom.

Screen Shot 2019-11-17 at 2 25 12 PM

- [x] update text to: `Please review the guidelines and rate the projects on at least one indexes. We recommend you not review the project on indexes you are not familiar with. <a>View Guidelines</a>` link in translation, make the text in link green.

- [x] use `name` and `organization` defined like this:

```php
        return [
            'userProfileId'   => $userProfile->getId(),
            'treecounterSlug' => $userProfile->getTreecounter()->getSlug(),
            'avatar'          => $userProfile->getImage(),
            'name'            => $userProfile->getFullname(),
            'organization'    => $userProfile->getReviewer()->getOrganization()
        ];
```
  
Screen Shot 2019-11-17 at 2 58 06 PM

- [x] API now returns associated plant project on create/update/delete, make sure to merge into redux.
Check out API docs https://staging.trilliontreecampaign.org/api/doc/review

------
Hi, @sagararyal please check this PR if that builds for you, I merged develop into old reviews pr.

### CODE REVIEW

**Comment by mahmed0715** [comment]
Duplicate of https://github.com/Plant-for-the-Planet-org/treecounter-app/pull/1326

**Comment by mahmed0715** [comment]
@sagararyal Review image functionality added.

File: app/config/index.js.dist [hunk 1]
```diff
@@ -11,7 +11,7 @@ export const initialProps = {
 
 export const context = {
   scheme: 'https', // API server protocol
-  host: 'test.trilliontreecampaign.org', // API server domain
+  host: 'staging.trilliontreecampaign.org', // API server domain
```
Comment (norbertschuler) [inline app/config/index.js.dist]: Do we want to change the default test server to staging?

File: ios/TreecounterApp.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -506,6 +523,24 @@
...
+		B6420F8A93A04D53AB95AB3B /* Zocial.ttf */ = {isa = PBXFileReference; name = "Zocial.ttf"; path = "../node_modules/react-native-vector-icons/Fonts/Zocial.ttf"; sourceTree = "<group>"; fileEncoding = undefined; lastKnownFileType = unknown; explicitFileType = undefined; includeInIndex = 0; };
+		CD8B5F1DC18E4D24ADEFF1C0 /* RNPaypalWrapper.xcodeproj */ = {isa = PBXFileReference; name = "RNPaypalWrapper.xcodeproj"; path = "../node_modules/react-native-paypal-wrapper/ios/RNPaypalWrapper.xcodeproj"; sourceTree = "<group>"; fileEncoding = undefined; lastKnownFileType = wrapper.pb-project; explicitFileType = undefined; includeInIndex = 0; };
+		B8FCC4013CD84444B8E5591B /* libRNPaypalWrapper.a */ = {isa = PBXFileReference; name = "libRNPaypalWrapper.a"; path = "libRNPaypalWrapper.a"; sourceTree = "<group>"; fileEncoding = undefined; lastKnownFileType = archive.ar; explicitFileType = undefined; includeInIndex = 0; };
```
Comment (norbertschuler) [inline ios/TreecounterApp.xcodeproj/project.pbxproj]: This deadly stuff for our iOS app (will be rejected in review process) has to be cleaned from the project again - once it got added from `react-native link` you either have to remove it manually or restart your pull request with the clean copy from e.g. the `develop branch` (merge it to this branch!) and then run `react-native link` again as I now removed the paypal module and this should not be reinserted again here!
I won't  comment on all the following lines where the paypal libs also occur again, but there are a lot more places to fix.

File: ios/TreecounterApp/Info.plist [hunk 3]
```diff
@@ -90,6 +90,22 @@
...
 		<string>OpenSans-SemiBold.ttf</string>
 		<string>OpenSans-SemiBoldItalic.ttf</string>
+		<string>AntDesign.ttf</string>
+		<string>Entypo.ttf</string>
+		<string>EvilIcons.ttf</string>
+		<string>Feather.ttf</string>
+		<string>FontAwesome.ttf</string>
+		<string>FontAwesome5_Brands.ttf</string>
+		<string>FontAwesome5_Regular.ttf</string>
+		<string>FontAwesome5_Solid.ttf</string>
```
Comment (norbertschuler) [inline ios/TreecounterApp/Info.plist]: I thought we limit this using only `FontAwesome`, but I assume all the fonts get added everytime again we use `react-native link`, so if we use `react-native-vector-icons` we are doomed to add all these unused fonts to our native projects!

**Comment by mahmed0715** [comment]
Is it possible @norbertschuler please fix ios part.

**Comment by mahmed0715** [comment]
It works on ios @norbertschuler ?

**Comment by norbertschuler** [comment]
Deleting a review does not work for me. The API call isn't even triggered when I choose the option:

Simulator Screen Shot - iPhone 11 - 2019-11-12 at 10 16 18

**Comment by norbertschuler** [comment]
On iOS the "Upload report" element is cut (on Android it is fine):

Simulator Screen Shot - iPhone 11 - 2019-11-12 at 10 15 51

**Comment by norbertschuler** [comment]
- [ ] an uploaded PDF is not shown when editing a review, so you don't know if you already uploaded a pdf (you can only see it from the "View PDF" button in the list of reviews.
- [ ] an uploaded image is not yet used anywhere, is it?

**Comment by mahmed0715** [comment]
The others issue I am checking on it
"an uploaded image is not yet used anywhere, is it?": Yeah the design doesnt have anything to show the images...
For the seelcted PDF I think I have added a text like "1 Pdf selected" when you select a pdf(before upload/submit form). you can not see that?

**Comment by mahmed0715** [comment]
isnt it enought to show "1 PDF file selected" when you select a pdf?

**Comment by norbertschuler** [comment]
> For the seelcted PDF I think I have added a text like "1 Pdf selected" when you select a pdf(before upload/submit form). you can not see that?

Yes, before I upload a selected PDF it shows:
Bildschirmfoto 2019-11-12 um 11 09 08
But if I edit the review after the upload once again it only shows this:
Bildschirmfoto 2019-11-12 um 11 09 35
So you just do not know if you already have a PDF attached to this review when you edit it.

**Comment by mahmed0715** [comment]
YEah in that case I think I shoudl show like "1 PDF file already upload" or like that. but if user upload new pdf it will overwrite the old one.

**Comment by norbertschuler** [comment]
> Yeah the design doesnt have anything to show the images...

@sagararyal image upload without using it anywhere is useless. What does the design say here?

**Comment by mahmed0715** [comment]
@norbertschuler @sagararyal I guess there should be a horizontal scroll area image layout to show images uploaded for a review like we do for project..

**Comment by sagararyal** [comment]
@mahmed0715 you can use the library @harshvitra1412 Harsh is using in competitions for the images.

**Comment by mahmed0715** [comment]
@norbertschuler @sagararyal Reviews should look good now. The Requested changes are applied.
The images are not required to show for now Sagar said.
So @norbertschuler please test according..

File: package.json [hunk 4]
```diff
@@ -13,6 +13,7 @@
     "lint": "eslint app",
     "lint:errors": "eslint app --quiet",
     "lint:json": "eslint . --ext .json",
+    "shake": "adb shell input keyevent 82",
```
Comment (norbertschuler) [inline package.json]: Is this command crucial for your daily work? Then let's keep it. I do not need it.

Comment (mahmed0715) [reply package.json]: Yeah , when I shake real device it gets disconnected, thats annoying, thus I thought this could be useful for other as well..

**Comment by norbertschuler** [comment]
I was able to create the following crash uploading an image and a PDF to one of my reviews on staging and then looking at the PDF - I would assume it's a bug in the PDF library we use. Please test yourself looking at my review from Nov 16th with the text "VIEWING THIS PDF CRASHES THE APP":
```
ReferenceError: ReferenceError: ReferenceError: Can't find variable: I18n
This error is located at:
    in PDFViewer (at SceneView.js:9)
    in SceneView (at StackViewLayout.js:478)
    in RCTView (at View.js:45)
    in View (at StackViewLayout.js:477)
    in RCTView (at View.js:45)
    in View (at StackViewLayout.js:476)
    in RCTView (at View.js:45)
    in View (at createAnimatedComponent.js:151)
    in AnimatedComponent (at screens.native.js:71)
    in Screen (at StackViewCard.js:42)
    in Card (at createPointerEventsContainer.js:26)
    in Container (at StackViewLayout.js:507)
    in RCTView (at View.js:45)
    in View (at screens.native.js:101)
    in ScreenContainer (at StackViewLayout.js:401)
    in RCTView (at View.js:45)
    in View (at StackViewLayout.js:400)
    in StackViewLayout (at withOrientation.js:30)
    in withOrientation (at StackView.js:49)
    in RCTView (at View.js:45)
    in View (at Transitioner.js:141)
    in Transitioner (at StackView.js:19)
    in StackView (at createNavigator.js:57)
    in Navigator (at createKeyboardAwareNavigator.js:11)
    in KeyboardAwareNavigator (at createNavigationContainer.js:376)
    in NavigationContainer (at SceneView.js:9)
    in SceneView (at DrawerView.js:87)
    in RCTView (at View.js:45)
    in View (at DrawerLayoutAndroid.android.js:198)
    in AndroidDrawerLayout (at DrawerLayoutAndroid.android.js:217)
    in DrawerLayoutAndroid (at DrawerView.js:84)
    in DrawerView (at createNavigator.js:57)
    in Navigator (at createNavigationContainer.js:376)
    in NavigationContainer (at AppDrawerNavigatorContainer.js:100)
    in RCTView (at View.js:45)
    in View (at AppDrawerNavigatorContainer.js:99)
    in AppDrawerNavigatorContainer (created by ConnectFunction)
    in ConnectFunction (at TreeCounter.native.js:29)
    in RCTView (at View.js:45)
    in View (at createAnimatedComponent.js:151)
    in AnimatedComponent (at react-native-safe-area-view/index.js:134)
    in SafeView (at withOrientation.js:54)
    in withOrientation (at TreeCounter.native.js:25)
    in App (created by ConnectFunction)
    in ConnectFunction (at App/index.native.js:19)
    in GlobalErrorBoundary (at App/index.native.js:18)
    in Provider (at App/index.native.js:17)
    in App (at renderApplication.js:35)
    in RCTView (at View.js:45)
    in View (at AppContainer.js:98)
    in ChildrenWrapper (at wrapRootComponent.js:9)
    in _class (at wrapRootComponent.js:8)
    in Root (at AppContainer.js:112)
    in RCTView (at View.js:45)
    in View (at AppContainer.js:115)
    in AppContainer (at renderApplication.js:34)
```

**Comment by mahmed0715** [comment]
Fixed those issues. . @norbertschuler

**Comment by mahmed0715** [comment]
how to fix ios/TreecounterApp.xcodeproj/project.pbxproj?

**Comment by norbertschuler** [comment]
> The images are not required to show for now Sagar said.

The uploaded image is also never shown again when editing the review. I wonder what the users thinks after uploading an image?

**Comment by mahmed0715** [comment]
@norbertschuler There is already added code to show existing image in edit mode, but looks like there issue with backend which cant send us images uploaded already.. Lets way for @jmiridis

**Comment by mahmed0715** [comment]
@sagararyal you said there should not be Reviews button on project details page if there is no existing reviews for that project.
But how to create review in this situation as there is no Reviews button there.

**Comment by norbertschuler** [comment]
It just came to my mind that while testing I never created a new review as I didn’t see any button or link to add one 🤔

**Comment by mahmed0715** [comment]
@norbertschuler please create some . Button is at the very bottom of the list.

**Comment by norbertschuler** [comment]
> @norbertschuler please create some . Button is at the very bottom of the list.

Works for me. Just again an error showing the uploaded PDF afterwards (my review named "brief"):

```
Can't find variable: url
_callee5$
    index.delta?platform=android&dev=true&minify=false:288530:66
tryCatch
    index.delta?platform=android&dev=true&minify=false:23350:23
invoke
    index.delta?platform=android&dev=true&minify=false:23525:32

```

**Comment by norbertschuler** [comment]
- [ ] Question: Why does opening the review screen triggers an API call `/suggest` for every single review displayed in the list?

Bildschirmfoto 2019-11-17 um 13 59 11

**Comment by mahmed0715** [comment]
@norbertschuler The suggest api is called to fetch the reviewer details. I asked @jmiridis to include some more info in reviwer object. that will fix this unnecessary api call. Another context is I see the suggest api never returns 304 status call(cache) but always 200(always newly fetched data)... We dont do caching?

**Comment by mahmed0715** [comment]
@norbertschuler pdf issue fixed.

**Comment by norbertschuler** [comment]
> Another context is I see the suggest api never returns 304 status(cache) but always 200(always newly fetched data)... We dont do caching?

This API is different from the rest. It's just a small PHP script tunneling the API requests to our Elastic server hosted at cloud.elastic.co, so the load on our server isn't that big as which other API calls connecting to the backend database. So for now I am fine doing some API requests on it.

**Comment by mahmed0715** [comment]
its good to test. Jorgo said all api changes should be good now.

**Comment by mahmed0715** [comment]
@sagararyal The requested changes are done on this PR. Please check...

**Comment by mahmed0715** [comment]
@sagararyal Here are some SS to show the fixes
Screenshot from 2019-11-20 23-18-32
Screenshot from 2019-11-20 23-18-51
Screenshot from 2019-11-20 23-19-08

**Comment by mahmed0715** [comment]
Sub task next to do https://github.com/Plant-for-the-Planet-org/treecounter-app/issues/1648

File: app/locales/de/plantProjectlabels.json [hunk 5]
```diff
@@ -18,5 +18,11 @@
   "view_more": "Weitere anschauen",
   "link": "Link",
   "exit": "Beenden",
-  "plant_project_update_success": "Planzprojekt erfolgreich aktualisiert"
+  "plant_project_update_success": "Planzungsprojekt erfolgreich aktualisiert",
```
Comment (sagararyal) [inline app/locales/de/plantProjectlabels.json]: I think these should come from the APIs, you should not include them here.

File: app/locales/de/reviews.json [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  "add_project_review": "Projektüberprüfung hinzufügen",
+  "guide_line": "Bitte überprüfen Sie die Richtlinien und bewerten Sie die Projekte in mindestens einem Index. Wir empfehlen Ihnen, das Projekt nicht anhand von Indizes zu überprüfen, mit denen Sie nicht vertraut sind.",
+  "brief_review": "KURZE ÜBERPRÜFUNG DES PROJEKTS",
+  "no_pdf_found": "Kein gültiges PDF gefunden!",
+  "reviews": "Bewertungen",
+  "summary_missing": "Kurzer Text fehlt",
+  "file_selected": "1 Datei ausgewählt",
+  "add_pictures": "BILDER HINZUFÜGEN",
+  "file_already_upload": "1 PDF-Datei bereits vorhanden",
+  "write_reviews": "Überprüfen Sie das Projekt",
```
Comment (sagararyal) [inline app/locales/de/reviews.json]: You don't need to add file in other locale, they will be ignored and updated by lingo hub.

File: app/locales/en/reviews.json [hunk 7]
```diff
@@ -0,0 +1,22 @@
+{
+  "community_reviews": "Expert Reviews",
+  "write_review": "Write a Review",
+  "view_pdf": "View PDF",
+  "edit_review": "Edit Review",
+  "delete_review": "Delete Review",
+  "land_quality": "Land Quality",
+  "co_benefits": "Co Benefits",
```
Comment (sagararyal) [inline app/locales/en/reviews.json]: remove land quality/ co-benefits all indexes from labels

File: app/locales/en/reviews.json [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+  "add_project_review": "Add Project Review",
+  "guide_line": "Please check the guidelines & only rate indexes you are most familiar with. All Indexes are optional.",
+  "brief_review": "BRIEF REVIEW OF PROJECT",
+  "no_pdf_found": "No valid pdf found!",
+  "reviews": "Reviews",
+  "summary_missing": "Brief text missing",
+  "file_selected": "1 File selected",
+  "add_pictures": "ADD PICTURES",
+  "file_already_upload": "1 PDF file already there",
+  "write_reviews": "Review the Project",
```
Comment (sagararyal) [inline app/locales/en/reviews.json]: duplicate of `write_review`

File: app/styles/selectplantproject/selectplantproject-full.native.js [hunk 9]
```diff
@@ -78,5 +78,19 @@ export default EStyleSheet.create({
     marginBottom: 10,
     marginLeft: 10,
     marginRight: 10
-  }
+  },
+  makePledgeButton: {
```
Comment (sagararyal) [inline app/styles/selectplantproject/selectplantproject-full.native.js]: maybe use a proper button name for reviews?

Comment (mahmed0715) [reply app/styles/selectplantproject/selectplantproject-full.native.js]: This button is not added by me and I am not using this, This has been used in only some of pledge related code. So its seems good to me.

File: app/locales/en/reviews.json (hunk 8 above)
Comment (mahmed0715) [reply app/locales/en/reviews.json]: deleted

File: app/locales/de/plantProjectlabels.json (hunk 5 above)
Comment (mahmed0715) [reply app/locales/de/plantProjectlabels.json]: okay from `/public/{version}/{_locale}/reviewIndexes/{translated}` api?
Confused! it still returned `"name": "review_index.name.survival_rate",` that needs translation resource to be present. Confusing to me.
Or on other PR we can make those things more dynamic?

File: app/locales/de/reviews.json (hunk 6 above)
Comment (mahmed0715) [reply app/locales/de/reviews.json]: fixed. removed this file.

File: app/locales/en/reviews.json (hunk 7 above)
Comment (mahmed0715) [reply app/locales/en/reviews.json]: removed these,
`
we get those from `/public/{version}/{_locale}/reviewIndexes/{translated}` api?
Confused! it still returned `"name": "review_index.name.survival_rate",` that needs translation resource to be present. Confusing to me.
Or on other PR we can make those things more dynamic?
`

**Comment by mahmed0715** [comment]
@sagararyal now We get reviewindexes from api . please check..

**Comment by mahmed0715** [comment]
@sagararyal some of the questions from @norbertschuler in the previous request are for you, like the design is okay or not.

**Comment by mahmed0715** [comment]
@norbertschuler your requests are fulfilled, the design is what we have currently, so no chnage on design. other issues fixed. Please test

**Comment by mahmed0715** [comment]
Currently working on ReviewIndexes that came from api.

**Comment by mahmed0715** [comment]
now all Good. I think if it works we can proceed to merge

**Comment by norbertschuler** [comment]
- [ ] I got a crash trying to edit my review:

```
TypeError: undefined is not an object (evaluating 'reviewIndexes[index].name')
This error is located at:
    in AddRatingSection (at AddReview.js:134)
    in RCTView (at View.js:45)
    in View (at AddReview.js:133)
    in RCTScrollContentView (at ScrollView.js:976)
    in RCTScrollView (at ScrollView.js:1115)
    in ScrollView (at AddReview.js:102)

```

File: app/components/Reviews/SingleReview.js [hunk 10]
```diff
@@ -0,0 +1,290 @@
...
+            review={review}
+            navigation={navigation}
+          />
+        </RBSheet>
+      </View>
+    );
+  }
+}
+
+const styles = StyleSheet.create({
```
Comment (norbertschuler) [inline app/components/Reviews/SingleReview.js]: Is there any special reason why these styles (also for all your other new components) have not been extracted into external style files as in our other modules to app/styles/<component>/...?

File: app/components/Reviews/AddReview/AddRatingSection.js [hunk 11]
```diff
@@ -0,0 +1,457 @@
...
+    };
+    this.setStateAndUpdateParent({ reviewIndexScores });
+    console.log('rating', rating, type);
+  }
+  render() {
+    let { reviewIndexes } = this.props;
+    let { reviewIndexScores } = this.state;
+    console.log('index scores', reviewIndexScores, reviewIndexes, this.props);
+    const guideLineUrl =
+      'https://startplanting.atlassian.net/wiki/spaces/PA/pages/25559041';
```
Comment (norbertschuler) [inline app/components/Reviews/AddReview/AddRatingSection.js]: @sagararyal do we really want to link to this page in our Atlassian Wiki?
Simulator Screen Shot - iPhone 11 - 2019-11-29 at 12 23 16

**Comment by norbertschuler** [comment]
- [ ] If you try sending a review without any score the backend API throws an error which is only shown as following to the user despite the API returns a clear error message `you must provide at least one score ` - we should either prevent sending the API request without any score given (my preferred solution) or present the error message to the user instead a general 400 error (hopefully the backend localized this error message):

Simulator Screen Shot - iPhone 11 - 2019-11-29 at 12 27 16

File: app/components/Reviews/SingleReview.js (hunk 10 above)
Comment (mahmed0715) [reply app/components/Reviews/SingleReview.js]: There is no special reason behind this, I will do that.

File: app/components/Reviews/AddReview/AddRatingSection.js (hunk 11 above)
Comment (mahmed0715) [reply app/components/Reviews/AddReview/AddRatingSection.js]: @sagararyal provided me this url.

**Comment by mahmed0715** [comment]
Yeah the form validation on front end and prevent api call for invalid is also my favorite. will check this..
@norbertschuler thanks for the catch..

File: app/components/Reviews/SingleReview.js (hunk 10 above)
Comment (mahmed0715) [reply app/components/Reviews/SingleReview.js]: done,

**Comment by mahmed0715** [comment]
@norbertschuler fixed those issues..

**Comment by norbertschuler** [comment]
> * [x]  I got a crash trying to edit my review:
> 
> ```
> TypeError: undefined is not an object (evaluating 'reviewIndexes[index].name')
> This error is located at:
>     in AddRatingSection (at AddReview.js:134)
>     in RCTView (at View.js:45)
>     in View (at AddReview.js:133)
>     in RCTScrollContentView (at ScrollView.js:976)
>     in RCTScrollView (at ScrollView.js:1115)
>     in ScrollView (at AddReview.js:102)
> ```

Reusing code is nice, but as you are using the same component for adding and editing a review both use case of cause have to be supported properly. I found two issues when editing a review:

- [ ] The title for editing an review is still showing "Add Project Review" which could irritate a user - you should add an additional resource and show them if using the component in "edit" mode

Simulator Screen Shot - iPhone 11 - 2019-11-29 at 16 42 01

- [ ] When deleting all the scores editing a review and then saving the review the app still crashes (s. below) while it does not crash any more when trying to add a new review with zero scores.

Simulator Screen Shot - iPhone 11 - 2019-11-29 at 16 38 41

**Comment by mahmed0715** [comment]
I suspect there is cache issue, please do a clear cache.
I triple checked the last issue and I could not find it, it works good.
Fixed all the issues @norbertschuler

**Comment by norbertschuler** [comment]
> I suspect there is cache issue, please do a clear cache.
> I triple checked the last issue and I could not find it, it works good.

I have to admit I do not know how exactly to reproduce this issue but if I try long enough editing reviews and setting and removing the scores to zero I still manage to create the case where I am able to submit the page with zero scores and therefore get the API error which is not caught and therefore showing the above error screen. Or maybe you can just think of the case where this might could happen?

File: app/components/Reviews/AddReview/AddReview.js [hunk 12]
```diff
@@ -0,0 +1,191 @@
...
+      review:
+        (props.navigation.state.params &&
+          props.navigation.state.params.review) ||
+        {}
+    };
+    this.create = this.create.bind(this);
+    this.onUpdate = this.onUpdate.bind(this);
+  }
+  onUpdate(data) {
+    this.setState({ review: data });
```
Comment (norbertschuler) [inline app/components/Reviews/AddReview/AddReview.js]: Setting the state of a component happens asynchronously. So if I am fast enough I can submit the button with empty scores!

File: app/components/Reviews/AddReview/AddReview.js [hunk 13]
```diff
@@ -0,0 +1,191 @@
...
+            plantProject,
+            summary,
+            reviewIndexScores,
+            pdfFile,
+            reviewImages
+          },
+          this.props.selectedPlantProject
+        );
+      } else {
+        await this.props.addReview(
```
Comment (norbertschuler) [inline app/components/Reviews/AddReview/AddReview.js]: In other components I saw using `dispatch()` for calling action independent from the UI. Here you use async and await. I am afraid you might get more bugs from concurrency issues this way.

File: app/components/Reviews/AddReview/AddReview.js (hunk 12 above)
Comment (mahmed0715) [reply app/components/Reviews/AddReview/AddReview.js]: Nope this is not the case, I found why you got 400 issue while edit mode, its because the score is a string {score:"0"} but we pressume it as number,

Comment (mahmed0715) [reply app/components/Reviews/AddReview/AddReview.js]: this doesnt happen when we create review as we initiate score as {score: 0}

File: app/components/Reviews/AddReview/AddReview.js (hunk 13 above)
Comment (mahmed0715) [reply app/components/Reviews/AddReview/AddReview.js]: Nope,,dont worry, this is common in es6 and its calling dispatch first and when we get the response from server, we are doing the update of redux store and at the end we return a promise(resolve)
thus you can use .then() or await if needed... if you dont need the response back in your component you dont need to return promise from the action thus you dont need to use await... 
this is common everywhere and also in our application as well..

File: app/components/Reviews/AddReview/AddReview.js (hunk 12 above)
Comment (norbertschuler) [reply app/components/Reviews/AddReview/AddReview.js]: Ok, I can not manage to reproduce the error any more.

**Comment by mahmed0715** [comment]
@norbertschuler @sagararyal https://github.com/Plant-for-the-Planet-org/treecounter-app/issues/1648  is the issue for Implementing image display on Reviews, I created couple of days ago.
